GHC 8 build failure

```
these derivations will be built:
  /nix/store/0nq3gqd2k2m3nvv8i81hmgsy6mjnz75i-shelly-1.6.5.drv
building path(s) ‘/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5’
setupCompilerEnvironmentPhase
Build with /nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204.
unpacking sources
unpacking source archive /nix/store/0my9vq4n0h7qb7hbq0wqwriii3xw911z-shelly-1.6.5.tar.gz
source root is shelly-1.6.5
setting SOURCE_DATE_EPOCH to timestamp 1449769415 of file shelly-1.6.5/ChangeLog.md
patching sources
Run jailbreak-cabal to lift version restrictions on build inputs.
compileBuildDriverPhase
setupCompileFlags: -package-db=/tmp/nix-build-shelly-1.6.5.drv-0/package.conf.d -j1 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /tmp/nix-build-shelly-1.6.5.drv-0/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5 --l
ibdir=$prefix/lib/$compiler --libsubdir=$pkgid --with-gcc=gcc --package-db=/tmp/nix-build-shell
y-1.6.5.drv-0/package.conf.d --ghc-option=-optl=-Wl,-rpath=/nix/store/yw95yx0j6d1qdacpqg2c26k76
wds4fnv-shelly-1.6.5/lib/ghc-8.0.0.20160204/shelly-1.6.5 --enable-split-objs --disable-library-
profiling --disable-profiling --enable-shared --enable-library-vanilla --enable-executable-dyna
mic --disable-tests
Configuring shelly-1.6.5...
Flags chosen: build-examples=False, lifted=False
Dependency async -any: using async-2.1.0
Dependency base >=4.6 && <5: using base-4.9.0.0
Dependency bytestring -any: using bytestring-0.10.7.0
Dependency containers -any: using containers-0.5.7.1
Dependency directory -any: using directory-1.2.5.0
Dependency enclosed-exceptions -any: using enclosed-exceptions-1.0.1.1
Dependency exceptions -any: using exceptions-0.8.2.1
Dependency lifted-async -any: using lifted-async-0.8.0.1
Dependency lifted-base -any: using lifted-base-0.2.3.6
Dependency monad-control -any: using monad-control-1.0.0.5
Dependency mtl -any: using mtl-2.2.1
Dependency process -any: using process-1.4.2.0
Dependency system-fileio -any: using system-fileio-0.3.16.3
Dependency system-filepath -any: using system-filepath-0.4.13.4
Dependency text -any: using text-1.2.2.0
Dependency time -any: using time-1.6
Dependency transformers -any: using transformers-0.5.1.0
Dependency transformers-base -any: using transformers-base-0.4.4
Dependency unix-compat -any: using unix-compat-0.4.1.4
Using Cabal-1.23.1.0 compiled by ghc-8.0
Using compiler: ghc-8.0.0.20160204
Using install prefix: /nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5
Binaries installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/bin
Libraries installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/lib/ghc-8.0.0.20160204/shelly-1.6.5
Private binaries installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/libexec
Data files installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/share/x86_64-linux-ghc-8.0.0.20160204/
shelly-1.6.5
Documentation installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/share/doc/x86_64-linux-ghc-8.0.0.20160
204/shelly-1.6.5
Configuration files installed in:
/nix/store/yw95yx0j6d1qdacpqg2c26k76wds4fnv-shelly-1.6.5/etc
No alex found
Using ar found on system at:
/nix/store/f9aj81zi2wam2dmnc30jhlr2lkwvgp49-binutils-2.23.1-dev/bin/ar
No c2hs found
No cpphs found
Using gcc version 4.9.3 given by user at:
/nix/store/2j5gi3bwsrpyfacivc28rlgiak2mdbxg-gcc-wrapper-4.9.3/bin/gcc
Using ghc version 8.0.0.20160204 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/ghc
Using ghc-pkg version 8.0.0.20160204 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.17.0 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/hpc
Using hsc2hs version 0.67 found on system at:
/nix/store/9ycqyik6sknsicdjhawn49yxv7m93mnc-ghc-8.0.0.20160204/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/yrj2if1z9k739b92n4la852nvzlkz1h2-hscolour-1.24/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/2j5gi3bwsrpyfacivc28rlgiak2mdbxg-gcc-wrapper-4.9.3/bin/ld
No lhc found
No lhc-pkg found
No pkg-config found
Using strip version 2.23 found on system at:
/nix/store/f9aj81zi2wam2dmnc30jhlr2lkwvgp49-binutils-2.23.1-dev/bin/strip
Using tar found on system at:
/nix/store/hmfmqrjv73f201z9al1sgsbzkih39cvj-gnutar-1.28/bin/tar
No uhc found
building
Building shelly-1.6.5...
Preprocessing library shelly-1.6.5...
[1 of 6] Compiling Shelly.Base      ( src/Shelly/Base.hs, dist/build/Shelly/Base.o )

src/Shelly/Base.hs:39:1: warning:
    The import of ‘ProcessHandle’
    from module ‘System.Process’ is redundant

src/Shelly/Base.hs:42:1: warning:
    The import of ‘liftM’ from module ‘Control.Monad’ is redundant
[2 of 6] Compiling Shelly.Find      ( src/Shelly/Find.hs, dist/build/Shelly/Find.o )

src/Shelly/Find.hs:67:5: warning:
    This binding for ‘traverse’ shadows the existing binding
      imported from ‘Prelude’ at src/Shelly/Find.hs:11:1-32
      (and originally defined in ‘Data.Traversable’)
[3 of 6] Compiling Shelly           ( src/Shelly.hs, dist/build/Shelly.o )

src/Shelly.hs:128:1: warning:
    The import of ‘Monoid’ from module ‘Data.Monoid’ is redundant

src/Shelly.hs:320:13: warning: Defined but not used: ‘fp’

src/Shelly.hs:375:5: warning:
    Pattern match(es) are non-exhaustive
    In an equation for ‘toHandle’:
        Patterns not matched: (Just NoStream)
[4 of 6] Compiling Shelly.Lifted    ( src/Shelly/Lifted.hs, dist/build/Shelly/Lifted.o )

src/Shelly/Lifted.hs:157:16: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:156:10
      Expected type: WriterT w m a
        Actual type: WriterT (forall a1. Monoid a1 => a1) m a
    • In the expression:
        WriterT
        $ do { a <- liftSh m;
               return (a, mempty) }
      In an equation for ‘liftSh’:
          liftSh m
            = WriterT
              $ do { a <- liftSh m;
                     return (a, mempty) }
      In the instance declaration for ‘MonadSh (WriterT w m)’
    • Relevant bindings include
        liftSh :: Sh a -> WriterT w m a
          (bound at src/Shelly/Lifted.hs:157:5)

src/Shelly/Lifted.hs:161:16: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:160:10
      Expected type: Strict.WriterT w m a
        Actual type: Strict.WriterT (forall a1. Monoid a1 => a1) m a
    • In the expression:
        Strict.WriterT
        $ do { a <- liftSh m;
               return (a, mempty) }
      In an equation for ‘liftSh’:
          liftSh m
            = Strict.WriterT
              $ do { a <- liftSh m;
                     return (a, mempty) }
      In the instance declaration for ‘MonadSh (Strict.WriterT w m)’
    • Relevant bindings include
        liftSh :: Sh a -> Strict.WriterT w m a
          (bound at src/Shelly/Lifted.hs:161:5)

src/Shelly/Lifted.hs:165:16: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:164:10
      Expected type: RWS.RWST r w s m a
        Actual type: RWS.RWST r (forall a1. Monoid a1 => a1) s m a
    • In the expression:
        RWS.RWST
        $ \ _ s
            -> do { a <- liftSh m;
                    return (a, s, mempty) }
      In an equation for ‘liftSh’:
          liftSh m
            = RWS.RWST
              $ \ _ s
                  -> do { a <- liftSh m;
                          .... }
      In the instance declaration for ‘MonadSh (RWS.RWST r w s m)’
    • Relevant bindings include
        liftSh :: Sh a -> RWS.RWST r w s m a
          (bound at src/Shelly/Lifted.hs:165:5)

src/Shelly/Lifted.hs:169:16: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:168:10
      Expected type: Strict.RWST r w s m a
        Actual type: Strict.RWST r (forall a1. Monoid a1 => a1) s m a
    • In the expression:
        Strict.RWST
        $ \ _ s
            -> do { a <- liftSh m;
                    return (a, s, mempty) }
      In an equation for ‘liftSh’:
          liftSh m
            = Strict.RWST
              $ \ _ s
                  -> do { a <- liftSh m;
                          .... }
      In the instance declaration for ‘MonadSh (Strict.RWST r w s m)’
    • Relevant bindings include
        liftSh :: Sh a -> Strict.RWST r w s m a
          (bound at src/Shelly/Lifted.hs:169:5)

src/Shelly/Lifted.hs:226:9: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:222:10
      Expected type: WriterT w m a
        Actual type: WriterT (forall a1. Monoid a1 => a1) m a
    • In the expression:
        WriterT
        $ liftM (\ x -> (x, mempty))
          $ liftShWith
            $ \ runInSh -> f $ \ k -> liftM WriterTShM $ runInSh $ runWriterT k
      In an equation for ‘liftShWith’:
          liftShWith f
            = WriterT
              $ liftM (\ x -> (x, mempty))
                $ liftShWith
                  $ \ runInSh -> f $ \ k -> liftM WriterTShM $ runInSh $ runWriterT k
      In the instance declaration for ‘MonadShControl (WriterT w m)’
    • Relevant bindings include
        f :: (forall x. WriterT w m x -> Sh (ShM (WriterT w m) x)) -> Sh a
          (bound at src/Shelly/Lifted.hs:225:16)
        liftShWith :: ((forall x.
                        WriterT w m x -> Sh (ShM (WriterT w m) x))
                       -> Sh a)
                      -> WriterT w m a
          (bound at src/Shelly/Lifted.hs:225:5)

src/Shelly/Lifted.hs:236:9: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:232:10
      Expected type: Strict.WriterT w m a
        Actual type: Strict.WriterT (forall a1. Monoid a1 => a1) m a
    • In the expression:
        Strict.WriterT
        $ liftM (\ x -> (x, mempty))
          $ liftShWith
            $ \ runInSh
                -> f $ \ k -> liftM StWriterTShM $ runInSh $ Strict.runWriterT k
      In an equation for ‘liftShWith’:
          liftShWith f
            = Strict.WriterT
              $ liftM (\ x -> (x, mempty))
                $ liftShWith
                  $ \ runInSh
                      -> f $ \ k -> liftM StWriterTShM $ runInSh $ Strict.runWriterT k
      In the instance declaration for
        ‘MonadShControl (Strict.WriterT w m)’
    • Relevant bindings include
        f :: (forall x.
              Strict.WriterT w m x -> Sh (ShM (Strict.WriterT w m) x))
             -> Sh a
          (bound at src/Shelly/Lifted.hs:235:16)
        liftShWith :: ((forall x.
                        Strict.WriterT w m x -> Sh (ShM (Strict.WriterT w m) x))
                       -> Sh a)
                      -> Strict.WriterT w m a
          (bound at src/Shelly/Lifted.hs:235:5)

src/Shelly/Lifted.hs:282:20: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:279:10
      Expected type: RWS.RWST r w s m a
        Actual type: RWS.RWST r (forall a1. Monoid a1 => a1) s m a
    • In the expression:
        RWS.RWST
        $ \ r s
            -> liftM (\ x -> (x, s, mempty))
               $ liftShWith $ \ runInSh -> f $ \ k -> ...
      In an equation for ‘liftShWith’:
          liftShWith f
            = RWS.RWST
              $ \ r s
                  -> liftM (\ x -> (x, s, mempty))
                     $ liftShWith $ \ runInSh -> f $ ...
      In the instance declaration for ‘MonadShControl (RWS.RWST r w s m)’
    • Relevant bindings include
        f :: (forall x.
              RWS.RWST r w s m x -> Sh (ShM (RWS.RWST r w s m) x))
             -> Sh a
          (bound at src/Shelly/Lifted.hs:282:16)
        liftShWith :: ((forall x.
                        RWS.RWST r w s m x -> Sh (ShM (RWS.RWST r w s m) x))
                       -> Sh a)
                      -> RWS.RWST r w s m a
          (bound at src/Shelly/Lifted.hs:282:5)

src/Shelly/Lifted.hs:292:20: error:
    • Couldn't match type ‘w’ with ‘forall a1. Monoid a1 => a1’
      ‘w’ is a rigid type variable bound by
        the instance declaration at src/Shelly/Lifted.hs:289:10
      Expected type: Strict.RWST r w s m a
        Actual type: Strict.RWST r (forall a1. Monoid a1 => a1) s m a
    • In the expression:
        Strict.RWST
        $ \ r s
            -> liftM (\ x -> (x, s, mempty))
               $ liftShWith $ \ runInSh -> f $ \ k -> ...
      In an equation for ‘liftShWith’:
          liftShWith f
            = Strict.RWST
              $ \ r s
                  -> liftM (\ x -> (x, s, mempty))
                     $ liftShWith $ \ runInSh -> f $ ...
      In the instance declaration for
        ‘MonadShControl (Strict.RWST r w s m)’
    • Relevant bindings include
        f :: (forall x.
              Strict.RWST r w s m x -> Sh (ShM (Strict.RWST r w s m) x))
             -> Sh a
          (bound at src/Shelly/Lifted.hs:292:16)
        liftShWith :: ((forall x.
                        Strict.RWST r w s m x -> Sh (ShM (Strict.RWST r w s m) x))
                       -> Sh a)
                      -> Strict.RWST r w s m a
          (bound at src/Shelly/Lifted.hs:292:5)
builder for ‘/nix/store/0nq3gqd2k2m3nvv8i81hmgsy6mjnz75i-shelly-1.6.5.drv’ failed with exit code 1
error: build of ‘/nix/store/0nq3gqd2k2m3nvv8i81hmgsy6mjnz75i-shelly-1.6.5.drv’ failed
```

--- a/src/Shelly/Lifted.hs
+++ b/src/Shelly/Lifted.hs
@@ -156,19 +156,19 @@ instance MonadSh m => MonadSh (Strict.StateT s m) where
 instance (Monoid w, MonadSh m) => MonadSh (WriterT w m) where
     liftSh m = WriterT $ do
         a <- liftSh m
-        return (a, mempty)
+        return (a, mempty :: w)
 instance (Monoid w, MonadSh m) => MonadSh (Strict.WriterT w m) where
     liftSh m = Strict.WriterT $ do
         a <- liftSh m
-        return (a, mempty)
+        return (a, mempty :: w)
 instance (Monoid w, MonadSh m) => MonadSh (RWS.RWST r w s m) where
     liftSh m = RWS.RWST $ \_ s -> do
         a <- liftSh m
-        return (a, s, mempty)
+        return (a, s, mempty :: w)
 instance (Monoid w, MonadSh m) => MonadSh (Strict.RWST r w s m) where
     liftSh m = Strict.RWST $ \_ s -> do
         a <- liftSh m
-        return (a, s, mempty)
+        return (a, s, mempty :: w)
 
 instance MonadSh m => S.ShellCmd (m Text) where
     cmdAll = (liftSh .) . S.run
@@ -223,7 +223,7 @@ instance (MonadShControl m, Monoid w)
          => MonadShControl (WriterT w m) where
     newtype ShM (WriterT w m) a = WriterTShM (ShM m (a, w))
     liftShWith f =
-        WriterT $ liftM  (\x -> (x, mempty)) $ liftShWith $ \runInSh -> f $ \k ->
+        WriterT $ liftM  (\x -> (x, mempty :: w)) $ liftShWith $ \runInSh -> f $ \k ->
             liftM WriterTShM $ runInSh $ runWriterT k
     restoreSh (WriterTShM m) = WriterT . restoreSh $ m
     {-# INLINE liftShWith #-}
@@ -233,7 +233,7 @@ instance (MonadShControl m, Monoid w)
          => MonadShControl (Strict.WriterT w m) where
     newtype ShM (Strict.WriterT w m) a = StWriterTShM (ShM m (a, w))
     liftShWith f =
-        Strict.WriterT $ liftM (\x -> (x, mempty)) $ liftShWith $ \runInSh -> f $ \k ->
+        Strict.WriterT $ liftM (\x -> (x, mempty :: w)) $ liftShWith $ \runInSh -> f $ \k ->
             liftM StWriterTShM $ runInSh $ Strict.runWriterT k
     restoreSh (StWriterTShM m) = Strict.WriterT . restoreSh $ m
     {-# INLINE liftShWith #-}
@@ -280,7 +280,7 @@ instance (MonadShControl m, Monoid w)
          => MonadShControl (RWS.RWST r w s m) where
     newtype ShM (RWS.RWST r w s m) a = RWSTShM (ShM m (a, s ,w))
     liftShWith f = RWS.RWST $ \r s ->
-        liftM (\x -> (x,s,mempty)) $ liftShWith $ \runInSh -> f $ \k ->
+        liftM (\x -> (x,s,mempty :: w)) $ liftShWith $ \runInSh -> f $ \k ->
             liftM RWSTShM $ runInSh $ RWS.runRWST k r s
     restoreSh (RWSTShM m) = RWS.RWST . const . const . restoreSh $ m
     {-# INLINE liftShWith #-}
@@ -290,7 +290,7 @@ instance (MonadShControl m, Monoid w)
          => MonadShControl (Strict.RWST r w s m) where
     newtype ShM (Strict.RWST r w s m) a = StRWSTShM (ShM m (a, s, w))
     liftShWith f = Strict.RWST $ \r s ->
-        liftM (\x -> (x,s,mempty)) $ liftShWith $ \runInSh -> f $ \k ->
+        liftM (\x -> (x,s,mempty :: w)) $ liftShWith $ \runInSh -> f $ \k ->
             liftM StRWSTShM $ runInSh $ Strict.runRWST k r s
     restoreSh (StRWSTShM m) = Strict.RWST . const . const . restoreSh $ m
     {-# INLINE liftShWith #-}

